auto_round/quantizer.py

```diff
@@ -343,6 +349,8 @@ def forward(self, x):
         from torch.functional import F
 
         weight = self.orig_layer.weight
+        if weight.device.type == 'meta':
```

Comment:
Can we encapsulate all code involving 'meta' or 'get_weight/bias' into functions? Additionally, can we combine the weight and bias sections? This will help reduce the likelihood of bugs when new features are merged, as many developers are not familiar with the existing logic and multiple changes increase the risk of errors.